src/server/game/Entities/Creature/Trainer.cpp

```diff
@@ -86,6 +89,18 @@ namespace Trainer
             player->CastSpell(player, trainerSpell->SpellId, true);
         else
             player->LearnSpell(trainerSpell->SpellId, false);
+
+        uint32 BattlePetSpell = trainerSpell->SpellId;
+        for (BattlePetSpeciesEntry const* entry : sBattlePetSpeciesStore)
+        {
+            if (entry->SummonSpellID == BattlePetSpell)
+            {
+                BattlePetMgr* battlePetMgr = player->GetSession()->GetBattlePetMgr();
+                battlePetMgr->AddPet(entry->ID, entry->CreatureID, BattlePetMgr::RollPetBreed(entry->ID), BattlePetMgr::GetDefaultPetQuality(entry->ID));
+                player->UpdateCriteria(CRITERIA_TYPE_OWN_BATTLE_PET_COUNT);
+                break;
```

Comment:
i think it would be better to move this code into Player::AddSpell on the place where mounts are handled too

Reply:
I agree with this